### PR TITLE
Add envVar CRUD methods

### PR DESCRIPTION
**Pull request by calufornia**
## Description

Adds `/dev/envVars/get|upsert|delete|list` endpoints. These endpoints will be called by the Studio UI to modify the `envVars` entries in the dev config file.

## Test plan
Tested in conjunction with web changes, unit tests

### CODE REVIEW

File: pkg/server/apidev/envvars.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package apidev
```
Comment (calufornia) [inline pkg/server/apidev/envvars.go:1]: As a side note, I'm going to incrementally pull things out of `api*` into their own files.

File: pkg/server/apidev/apidev.go [hunk 2]
```diff
@@ -37,6 +37,11 @@ func AttachDevRoutes(r *mux.Router, s *state.State) {
 	r.Handle("/info", handlers.Handler(s, GetInfoHandler)).Methods("GET", "OPTIONS")
 	r.Handle("/version", handlers.Handler(s, GetVersionHandler)).Methods("GET", "OPTIONS")
 
+	r.Handle("/envVars/get", handlers.Handler(s, GetEnvVarHandler)).Methods("GET", "OPTIONS")
+	r.Handle("/envVars/upsert", handlers.HandlerWithBody(s, UpsertEnvVarHandler)).Methods("POST", "OPTIONS")
+	r.Handle("/envVars/delete", handlers.HandlerWithBody(s, DeleteEnvVarHandler)).Methods("POST", "OPTIONS")
```
Comment (colinking) [inline pkg/server/apidev/apidev.go:42]:
```suggestion
	r.Handle("/envVars/delete", handlers.HandlerWithBody(s, DeleteEnvVarHandler)).Methods("DELETE", "OPTIONS")
```

File: pkg/server/apidev/apidev.go [hunk 3]
```diff
@@ -37,6 +37,11 @@ func AttachDevRoutes(r *mux.Router, s *state.State) {
 	r.Handle("/info", handlers.Handler(s, GetInfoHandler)).Methods("GET", "OPTIONS")
 	r.Handle("/version", handlers.Handler(s, GetVersionHandler)).Methods("GET", "OPTIONS")
 
+	r.Handle("/envVars/get", handlers.Handler(s, GetEnvVarHandler)).Methods("GET", "OPTIONS")
+	r.Handle("/envVars/upsert", handlers.HandlerWithBody(s, UpsertEnvVarHandler)).Methods("POST", "OPTIONS")
```
Comment (colinking) [inline pkg/server/apidev/apidev.go:41]:
```suggestion
	r.Handle("/envVars/upsert", handlers.HandlerWithBody(s, UpsertEnvVarHandler)).Methods("PUT", "OPTIONS")
```

File: pkg/server/apidev/envvars.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			},
+		}, nil
+	}
+
+	return GetEnvVarResponse{}, errors.Errorf("env var with name %s not found", name)
+}
+
+func UpsertEnvVarHandler(ctx context.Context, state *state.State, r *http.Request, req LocalEnvVar) (struct{}, error) {
+	if req.Name == "" {
+		return struct{}{}, errors.New("name cannot be empty")
```
Comment (colinking) [inline pkg/server/apidev/envvars.go:35]: We should do validation, too. e.g. https://stackoverflow.com/a/2821183/1935727

File: pkg/server/apidev/envvars.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	Value string `json:"value"`
+}
+
+type ListEnvVarsResponse struct {
+	EnvVars []LocalEnvVar `json:"envVars"`
+}
+
+func ListEnvVarsHandler(ctx context.Context, state *state.State, r *http.Request) (ListEnvVarsResponse, error) {
+	envVars := make([]LocalEnvVar, 0, len(state.DevConfig.EnvVars))
+	for name, val := range state.DevConfig.EnvVars {
```
Comment (colinking) [inline pkg/server/apidev/envvars.go:68]: Maybe sort by name so the return values are consistent?

**Review by colinking — APPROVED** [review]

**Pull request merged**